build-tools/automation/build.groovy

```diff
@@ -65,18 +83,48 @@ timestamps {
 
             def buildType = isPr ? 'PR' : 'CI'
 
+            echo "Git repo: ${env.GitRepo}"     // Defined as an environment variable in the jenkins build definition
             echo "Job: ${env.JOB_BASE_NAME}"
             echo "Branch: ${branch}"
             echo "Commit: ${commit}"
             echo "Build type: ${buildType}"
+
             if (isPr) {
                 echo "PR id: ${env.ghprbPullId}"
                 echo "PR link: ${env.ghprbPullLink}"
+
+                // Clear out the PR title and description. This is the equivalent of $JENKINS_HOME/global-pre-script/remove-problematic-ghprb-parameters.groovy used by freestyle builds
+                echo "Clearing the PR title and description environment variables to avoid any special characters contained within from tripping up the build"
+                env.ghprbPullTitle = ''
+                env.ghprbPullLongDescription = ''
+
+                if (hasPrLabel(env.GitRepo, env.ghprbPullId, 'full-mono-integration-build')) {
+                    buildTarget = 'jenkins'
+                } else {
+                    buildTarget = 'all'
+                }
             }
+
+            echo "${buildType} buildTarget: ${buildTarget}"
         }
 
-        stageWithTimeout('clean', 30, 'SECONDS', 'xamarin-android_tmp', true) {    // Typically takes less than a second
-            deleteDir()
+        stageWithTimeout('clean', 30, 'SECONDS', XADir, true) {    // Typically takes less than a second
+            def commandStatus = sh(
+                script: """
+                        # We need to make sure there's no test AVD present and that the Android emulator isn't running
+                        # This is to assure that all tests start from the same state
+                        env
+                        killall -9 qemu-system-x86_64 || true
+                        if [ -d "\$HOME/.android/avd/XamarinAndroidTestRunner.avd" ]; then
+                            rm -rf \$HOME/.android/avd/XamarinAndroidTestRunner.*
+                        fi
```

Comment:
I'd put these into separate `sh()` statements / groovy comments:

``` groovy
// We need to make sure there's no test AVD present and that the Android emulator isn't running
// This is to assure that all tests start from the same state
sh "env"
sh "killall -9 qemu-system-x86_64 || true"
def commandStatus = sh (script: "rm -rf \$HOME/.android/avd/XamarinAndroidTestRunner.*",  returnStatus: true)
```

Not sure if you really need the `env`.

Reply:
`env` *somewhere* is occasionally useful for bizarro-diagnostic purposes ("why's it work on machine *A* but not machine *B*?").

Having it *here* is a tad odd; the `checkout` stage would make more sense.

Reply:
Addressed with commit https://github.com/xamarin/xamarin-android/pull/2767/commits/bddddd19534ca861872c494736c0a56c83aed7f2. The `env` command has been moved to the checkout stage. Broke out commands into their own 'sh' execution.  `rm -rf` does not lead to an error if the directory does not exist and so removed the dir existence check.